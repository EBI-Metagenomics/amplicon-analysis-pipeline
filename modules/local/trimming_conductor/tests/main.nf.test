nextflow_process {

    name "Test Process TRIMMING_CONDUCTOR"
    script "../main.nf"
    process "TRIMMING_CONDUCTOR"
    tag "modules"
    tag "trimming_conductor"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    "16S-V3-V4",
                    file('${projectDir}/modules/local/trimming_conductor/tests/data/test_general_primer_out.txt', checkIfExists: true),
                    file('${projectDir}/modules/local/trimming_conductor/tests/data/test_std_primers.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.trimming_conductor_out).match("trimming_conductor_out") }
            )
        }

    }

}
