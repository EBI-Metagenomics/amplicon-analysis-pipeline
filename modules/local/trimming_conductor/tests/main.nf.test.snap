{
    "trimming_conductor_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "16S-V3-V4",
                    "test_trimming_conductor_out_16S-V3-V4.txt:md5,04efd337d1242cb69b780f744b73cb97"
                ]
            ]
        ],
        "timestamp": "2023-11-03T13:44:36.708347748"
    }
}